tests/GenerateScriptTests/GenerateScriptTests.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace GenerateScriptTests;
+
+public class GenerateScriptTests
+{
+    public enum PackageType
+    {
+        Reference,
+        Text
+    }
+    public static IEnumerable<object[]> Data => new List<object[]>
+    {
+        new object[] { "Microsoft.CodeAnalysis.CSharp.Workspaces", "4.5.0", PackageType.Reference },
+        new object[] { "System.Threading.Tasks.Extensions", "4.5.4", PackageType.Reference },
+        new object[] { "System.Diagnostics.DiagnosticSource", "7.0.2", PackageType.Reference },
+        new object[] { "Microsoft.CodeAnalysis", "4.5.0", PackageType.Text },
+    };
+    
+    public string SandboxDirectory { get; set; }
+    public string RepoRoot { get; set; }
+    public ITestOutputHelper output { get; set; }
+
+    public GenerateScriptTests(ITestOutputHelper output)
+    {
+        this.output = output;
+        RepoRoot = Environment.CurrentDirectory.Substring(0, Environment.CurrentDirectory.IndexOf("source-build-reference-packages") + 31);
+        SandboxDirectory = Path.Combine(Environment.CurrentDirectory, $"GenerateTests-{DateTimeOffset.UtcNow.ToUnixTimeSeconds().ToString()}");
+        Directory.CreateDirectory(SandboxDirectory);
+    }
+
+    [Theory]    
+    [MemberData(nameof(GenerateScriptTests.Data), MemberType = typeof(GenerateScriptTests))]
+    public void VerifyGenerateScript(string package, string version, PackageType type)
+    {
+        string command = Path.Combine(RepoRoot, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "generate.cmd" : "generate.sh");
+        string arguments = string.Format("-p {0} -x -d {1}",string.Format("{0},{1}", package, version), SandboxDirectory);
```

Comment:
String interpolation would simplify this

```suggestion
        string arguments = $"-p {package},{version{ -x -d {SandboxDirectory}";
```